File: java-app/src/main/java/com/mercadolibre/java_app/review/ReviewActivity.java
package com.mercadolibre.java_app.review;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.ColorRes;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.mercadolibre.java_app.R;
import com.mercadolibre.java_app.commons.Purchase;
import com.mercadolibre.java_app.congrats.CongratsActivity;
import com.mercadolibre.java_app.review.presenter.ReviewPresenter;
import com.mercadolibre.java_app.services.shipping.ShippingOptionServices;

public class ReviewActivity extends AppCompatActivity implements ReviewContract.ReviewView {
    private ReviewPresenter presenter;
    private Button confirmButton;
    private TextView productPrice;
    private TextView totalPrice;
    private TextView shippingPrice;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.review_layout);

        presenter = new ReviewPresenter(this, new ShippingOptionServices());

        setUpControls();
        setUpEvents();
        presenter.calculatePrices();
    }

    private void setUpControls() {
        this.confirmButton = findViewById(R.id.confirm_button);
        this.productPrice = findViewById(R.id.producto_price);
        this.totalPrice = findViewById(R.id.total_price);
        this.shippingPrice = findViewById(R.id.shipping_price);
    }

    private void setUpEvents() {
        confirmButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                presenter.validatePurchase();
            }
        });
    }

    @Override
    public void showProductPrice(String price) {
        productPrice.setText(price);
    }

    @Override
    public void showTotalPrice(String price) {
        totalPrice.setText(price);
    }

    @Override
    public void showShippingPrice(@NonNull String price, @ColorRes int color) {
        shippingPrice.setText(price);
        shippingPrice.setTextColor(ContextCompat.getColor(this, color));
    }

    @Override
    public void showShippingType(@NonNull String shippingType) {}

    @Override
    public void goToCongrats(Purchase purchase) {
        Intent intent = new Intent(this, CongratsActivity.class);
        intent.putExtra("purchase", purchase);

        startActivity(intent);
    }
}

File: kt-app/src/main/java/com/mercadolibre/melikotlinmeetup/review/ReviewActivity.kt
package com.mercadolibre.melikotlinmeetup.review

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.content.ContextCompat
import com.mercadolibre.melikotlinmeetup.R
import com.mercadolibre.melikotlinmeetup.commons.Purchase
import com.mercadolibre.melikotlinmeetup.congrats.CongratsActivity
import com.mercadolibre.melikotlinmeetup.review.presenter.ReviewPresenter
import com.mercadolibre.melikotlinmeetup.services.shipping.ShippingOptionServices
import kotlinx.android.synthetic.main.review_layout.*

class ReviewActivity : AppCompatActivity(), ReviewView {
    private lateinit var presenter: ReviewPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.review_layout)

        presenter = ReviewPresenter(this, ShippingOptionServices())
        presenter.calculatePrices()

        setUpControls()
    }

    private fun setUpControls() {
        confirm_button.setOnClickListener { presenter.validatePurchase() }
    }

    override fun showProductPrice(price: String) {
        producto_price.text = price
    }

    override fun showShippingPrice(price: String, color: Int) {
        shipping_price.text = price
        shipping_price.setTextColor(ContextCompat.getColor(this, color))
    }

    override fun showTotalPrice(price: String) {
        total_price.text = price
    }

    override fun showShippingType(shippingType: String) {}

    override fun goToCongrats(purchase: Purchase) {
        val intent = Intent(this, CongratsActivity::class.java)
        intent.putExtra("purchase", purchase)

        startActivity(intent)
    }
}
